lib/client.js

```diff
@@ -1082,12 +1085,16 @@ MqttClient.prototype._handlePubrel = function (packet, callback) {
   that.incomingStore.get(packet, function (err, pub) {
     if (!err && pub.cmd !== 'pubrel') {
       that.emit('message', pub.topic, pub.payload, pub)
-      that.incomingStore.put(packet)
-      that.handleMessage(pub, function (err) {
+      that.incomingStore.put(packet, function (err) {
```

Comment:
I'd prefer if we called `handleMessage` _before_ sending a PUBCOMP.

Reply:
The updated code calls `handleMessage` before sending a PUBCOMP. The sequence is not changed before/after update.